build.gradle

```diff
@@ -322,10 +322,14 @@ task updateVersion {
             }
         }
         ant.replaceregexp(file:'build.gradle', match: '"opensearch.version", "\\d.*"', replace: '"opensearch.version", "' + newVersion.tokenize('-')[0] + '-SNAPSHOT"', flags:'g', byline:true)
-        // Extract the oldBWCVersion from the existing OpenSearch Version (oldBWCVersion = major + (minor-1) + patch)
+
         ext.os_version_without_snapshot = opensearch_version.tokenize('-')[0]
-        ext.oldBWCVersion = os_version_without_snapshot.tokenize('.')[0] + '.' + Integer.toString(Integer.valueOf(os_version_without_snapshot.tokenize('.')[1]) - 1) + '.' + os_version_without_snapshot.tokenize('.')[2]
-        // Include the current OpenSearch Version before version bump to the bwc_version matrix
-        ant.replaceregexp(file:".github/workflows/backwards_compatibility_tests_workflow.yml", match: oldBWCVersion, replace: oldBWCVersion + '", "' + opensearch_version.tokenize('-')[0], flags:'g', byline:true)
+        // This condition will check if the BWC workflow is already updated or not and will run next steps if not updated
+        if (!fileTree(".github/workflows/backwards_compatibility_tests_workflow.yml").getSingleFile().text.contains(os_version_without_snapshot)) {
+            // Extract the oldBWCVersion from the existing OpenSearch Version (oldBWCVersion = major + (minor-1) + patch)
+            ext.oldBWCVersion = os_version_without_snapshot.tokenize('.')[0] + '.' + Integer.toString(Integer.valueOf(os_version_without_snapshot.tokenize('.')[1]) - 1) + '.' + os_version_without_snapshot.tokenize('.')[2]
```

Comment:
nit: This is a bit hard to read. Can we break this up into a couple of lines with temp variables?

Reply:
ack

Reply:
seems that we're finding match with version string anywhere in the file. Is it possible to make match condition more strict, say make sure that line with that matches starts with "bwc_version :"?

Reply:
no, not required. We don't need to search in a given line because I'm looking for opensearch version.
For example - In the 2.x branch which has OS version as 2.7.0, if I'm trying to update OS version to 2.7.0 again, then it will look for 2.7.0 in BWC workflow and it will find it and fail the condition.
If we are trying to update to 2.8.0, then it will look for the OS version before increment which is 2.7.0 in BWC workflow and it can't find it because the OS version is already incremented to 2.8.0 before reaching this if condition. So, it will update the BWC matrix.

Reply:
I mean a bit different case, say opensearch main version has been updated from 2.6.0 to 2.7.0-SNAPSHOT, is it possible that this PR executed before this update and because it finds 2.6.0 (as base version, not the bwc) then it skips the update?
I think it makes sense if updates for each version happen independently, and if infra script executed once, but not sure if that's the case here